x-pack/plugins/viz_editor/public/pseudo_plugins/xy_chart_plugin/yaxis_editor.tsx

```diff
@@ -30,45 +29,28 @@ export function YAxisEditor({
     onChangeVisModel(updateColumn(col, operation, visModel));
   };
 
-  const onChangeOperation = (e: React.ChangeEvent<HTMLSelectElement>) => {
-    const operationName = e.target.value;
-    const operation = {
-      ...currentOperation,
-      operation: operationName,
-    };
+  const column = selectColumn(col, visModel);
 
-    onChangeVisModel(updateColumn(col, operation, visModel));
-  };
-
-  const options = [
-    {
-      value: 'count',
-      text: 'Count',
-    },
-    {
-      value: 'avg',
-      text: `Average of ${fieldName}`,
-    },
-    {
-      value: 'sum',
-      text: `Sum of ${fieldName}`,
-    },
-  ];
+  if (!column) {
+    // TODO...
+    return <span>N/A</span>;
+  }
 
   return (
     <Draggable
       canHandleDrop={(f: DatasourceField) => f && f.type === 'number'}
       onDrop={onDropField}
     >
-      {fieldName ? (
-        <EuiSelect
-          options={options}
-          value={currentOperation.operation}
-          onChange={onChangeOperation}
-        />
-      ) : (
-        columnSummary(selectColumn(col, visModel))
-      )}
+      <OperationEditor
+        column={column}
+        visModel={visModel}
+        onColumnChange={newColumn => {
+          onChangeVisModel(updateColumn(col, newColumn, visModel));
+        }}
+        allowedOperations={['avg', 'cardinality', 'count', 'sum', 'window']}
```

Comment:
I'm pretty worried about the maintainability of this approach- string types are already hard to manage, and now we're expecting that if we ever add types, we will need to update all the editors?

Is there a way to do something like an Operations Registry?

```
registerOperation('cardinality', {
  supportedDataTypes: DataTypesRegistry.ALL,
  // more config here
});

registerOperation('window', {
  supportedDataTypes: DataTypesRegistry.NUMERIC,
  additionalArguments: {
    windowSize: 5,
  },
  // more config here
});
```

Reply:
If we have some kind of dynamic registry, we loose typing support, right? I wonder whether we can get away with a fixed set of operations which each editor has to use (maybe with an escape hatch `esagg` which just has a direct configuration object for esaggs as an argument) and just use a helper function from the common folder like `allowedOperations={getNumericOperations()}`.

Reply:
I agree that we should be using a static registry up until it breaks- my concern is specifically that this will break as soon as we want to support rollups. I suspect we'll need to discuss this issue more to come to a conclusion.